upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template

```diff
@@ -1059,113 +1046,161 @@ spec:
       - key: node-role.kubernetes.io/control-plane
         operator: Exists
       topologySpreadConstraints:
-      - maxSkew: 1
-        topologyKey: topology.kubernetes.io/zone
-        whenUnsatisfiable: ScheduleAnyway
-        labelSelector:
-          matchLabels:
-            karpenter: webhook
-      - maxSkew: 1
```

Comment:
Why is this one being removed?

Reply:
The last version of Karpenter merges the controller and the webhook into the same container.
The new `topologySpreadConstraints` is now (based ib the default output of the helm chart):
https://github.com/kubernetes/kops/blob/dee14fb84330644298db6daa3c18840954b963f7/upup/models/cloudup/resources/addons/karpenter.sh/k8s-1.19.yaml.template#L1048-L1056

Reply:
The block you removed is one we use for all addons so it shouldn't be removed just because upstream doesn't have it.

Reply:
Well, we should decide which selector we want to use as Karpenter is a single pod now:
```
karpenter: webhook
```
or
```
karpenter: controller
```
?

The `update-expected` script seems to accept only 1 label "karpenter".

Reply:
Not sure what you mean what update-expected accepts., It only renders the manifests. These labels need to select the appropriate pods. Usually the same selector one uses for the Deployment selector (and things like PDB).
I am guessing the `karpenter` label isn't required as part of the selector at all.

Reply:
I have added the webhook label in this commit https://github.com/kubernetes/kops/pull/15144/commits/8baad20aca0d9db2dc7bdbb7aa637d31be969a5e but the update-expected script only keeps the `karpenter: webhook` label in tests files.

Reply:
Just to be clear, it's the entry using hostname as topology key that is missing. Not selecting specifically for the webhook label.

Reply:
Haaaaaaaaa ^^'
Sorry I didn't understand that. Let me fix that. :ok_hand:

Reply:
Fixed in https://github.com/kubernetes/kops/pull/15144/commits/6a76b51cdaef8c2cbdb79eb43ddc1f35f4d7f2e7 :+1: